### PR TITLE
feat(fa): Student form added to application

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# Student form

https://app.asana.com/0/1201600280110882/1201624216643399

## What

 - asking applicant if they are students 
 - if so what kind of education are they receiving 

## Why

 - Mandatory question to work on application

## Screenshots / Gifs
Screenshot 2022-01-11 at 13 58 04

Screenshot 2022-01-11 at 13 57 54

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          ]}
+          largeButtons
+          backgroundColor="white"
+          error={typeInput.error}
+        />
+      </Box>
+      {getValues(typeInput.id) === TwoTypeAnswers.Yes && (
+        <Box>
+          <InputController
+            id={customeInput.id}
```
Comment (MargretFinnboga) [inline libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx:47]: waiting for homecircumstance PR to do this via classnames

File: libs/application/templates/financial-aid/src/lib/types.ts [hunk 2]
```diff
@@ -5,6 +5,11 @@ export enum DataProviderTypes {
   NationalRegistry = 'NationalRegistryProvider',
 }
 
+export enum TwoTypeAnswers {
```
Comment (Rikkiabb) [inline libs/application/templates/financial-aid/src/lib/types.ts:8]: Can we renam this? Something like ApproveOptions or along that line would be better

File: libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+const StudentForm = ({ errors, application }: FAFieldBaseProps) => {
+  const { formatMessage } = useIntl()
+  const { answers } = application
+
+  const typeInput = {
+    id: 'student.isStudent',
+    error: errors?.student?.isStudent,
+  }
+  const customeInput = {
```
Comment (Rikkiabb) [inline libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx:17]: typo here

File: libs/application/templates/financial-aid/src/lib/messages/studentForm.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { defineMessages } from 'react-intl'
+
+export const studentForm = {
+  general: defineMessages({
+    sectionTitle: {
+      id: 'fa.application:section.studentForm.general.sectionTitle',
+      defaultMessage: 'Nám',
+      description: 'Student form Page Title',
```
Comment (Rikkiabb) [inline libs/application/templates/financial-aid/src/lib/messages/studentForm.ts:8]: Wrong description here

**Review by Rikkiabb — CHANGES_REQUESTED** [review]
minor stuff

File: libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx (hunk 1 above)
Comment (Rikkiabb) [reply libs/application/templates/financial-aid/src/fields/StudentForm/studentForm.tsx:47]: it is approved

File: libs/application/templates/financial-aid/src/lib/messages/studentForm.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    pageTitle: {
+      id: 'fa.application:section.studentForm.general.pageTitle',
+      defaultMessage: 'Ertu í lánshæfu námi?',
+      description: 'Student form page title',
+    },
+  }),
+  form: defineMessages({
+    no: {
+      id: 'fa.application:section.studentForm.form.notStudent',
```
Comment (Rikkiabb) [inline libs/application/templates/financial-aid/src/lib/messages/studentForm.ts:18]: the id should match the name here and on

File: libs/application/templates/financial-aid/src/lib/messages/studentForm.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      defaultMessage: 'Nei',
+      description:
+        'Student form radio selection when applicant is not a student',
+    },
+    yes: {
+      id: 'fa.application:section.studentForm.form.isStudent',
```
Comment (Rikkiabb) [inline libs/application/templates/financial-aid/src/lib/messages/studentForm.ts:24]: and here

**Review by Rikkiabb — CHANGES_REQUESTED** [review]

**Review by Rikkiabb — APPROVED** [review]
👌

**Pull request merged**